File: crates/deepsec-cli/src/commands/init_project.rs
use crate::config_loader::Context;
use anyhow::Result;
use clap::Args as ClapArgs;
use deepsec_core::ensure_project;

#[derive(Debug, ClapArgs)]
pub struct Args {
    #[arg(long)]
    pub project_id: String,
    #[arg(long, default_value = ".")]
    pub root: String,
    #[arg(long)]
    pub github_url: Option<String>,
}

pub fn run(args: Args, ctx: &Context) -> Result<()> {
    let cfg = ensure_project(
        &ctx.data_root,
        &args.project_id,
        &args.root,
        args.github_url.as_deref(),
    )?;
    println!(
        "Initialized project '{}' at {}",
        cfg.project_id, cfg.root_path
    );
    Ok(())
}

File: internal/cli/commands/init_project.go
package commands

import (
	"fmt"

	"github.com/noeljackson/deepsec/internal/cli"
	"github.com/spf13/cobra"
)

// NewInitProjectCmd writes `data/<id>/project.json` for a new project.
func NewInitProjectCmd(loader func() (*cli.Context, error)) *cobra.Command {
	var projectID, root, githubURL string
	cmd := &cobra.Command{
		Use:   "init-project",
		Short: "Initialize a new project entry in `data/`",
		RunE: func(cmd *cobra.Command, _ []string) error {
			ctx, err := loader()
			if err != nil {
				return err
			}
			cfg, err := ctx.DataRoot.EnsureProject(projectID, root, githubURL)
			if err != nil {
				return err
			}
			fmt.Printf("Initialized project %q at %s\n", cfg.ProjectID, cfg.RootPath)
			return nil
		},
	}
	cmd.Flags().StringVar(&projectID, "project-id", "", "Project id")
	cmd.Flags().StringVar(&root, "root", ".", "Project root path")
	cmd.Flags().StringVar(&githubURL, "github-url", "", "GitHub blob URL for line links")
	_ = cmd.MarkFlagRequired("project-id")
	return cmd
}
